File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> m;
        for(auto str:strs){
            string tmp = str;
            sort(tmp.begin(),tmp.end());
            m[tmp].push_back(str);
        }
        for(auto vec:m){
            res.push_back(vec.second);
        }
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        table = {};
        for str in strs:
            tmp = ''.join(sorted(str))
            # table.setdefault(tmp, []).append(str)
            if tmp not in table:
                table[tmp] = [str]
            else:
                table[tmp].append(str)
        result = []
        for key in table:
            result.append(table[key])
        return result